### PR TITLE
Make NewCC compilable (again)

### PR DESCRIPTION
**Pull request by woachk**
A light fix that is needed before doing anything else.

### CODE REVIEW

File: ntoskrnl/cache/mdlsup.c [hunk 1]
```diff
@@ -106,13 +106,14 @@ CcMdlWriteComplete(IN PFILE_OBJECT FileObject,
...
 
+/* This API call is currently unimplemented */
 VOID
 NTAPI
 CcMdlWriteComplete2(IN PFILE_OBJECT FileObject,
                     IN PLARGE_INTEGER FileOffset,
                     IN PMDL MdlChain)
 {
-    UNIMPLEMENTED
+
```
Comment (HBelusca) [inline ntoskrnl/cache/mdlsup.c]: Fixes:
"UNIMPLEMENTED;"
(with a semi-colon).

File: ntoskrnl/cache/mdlsup.c [hunk 2]
```diff
@@ -77,13 +77,13 @@ CcMdlReadComplete(IN PFILE_OBJECT FileObject,
...
 }
-
+/* This API call is currently unimplemented */
 VOID
 NTAPI
 CcMdlReadComplete2(IN PFILE_OBJECT FileObject,
                    IN PMDL MdlChain)
 {
-    UNIMPLEMENTED
+	return;
```
Comment (HBelusca) [inline ntoskrnl/cache/mdlsup.c]: "UNIMPLEMENTED;" (with a semi-colon).

File: ntoskrnl/cache/mdlsup.c [hunk 3]
```diff
@@ -77,13 +77,13 @@ CcMdlReadComplete(IN PFILE_OBJECT FileObject,
 {
     IoFreeMdl(MdlChain);
 }
-
+/* This API call is currently unimplemented */
```
Comment (HBelusca) [inline ntoskrnl/cache/mdlsup.c]: Then, no need to put the comment here :P

**Comment by binarymaster** [comment]
Please use your full name and correct e-mail according to [contribution rules](https://github.com/reactos/reactos/blob/master/CONTRIBUTING.md#rules-and-recommendations). Thanks.

Currently there is `someone <someone@desktop-j4sggtq>`.

**Comment by tkreuzer** [comment]
Is this still a thing? If you want to revive it, start by making the code somewhat readable. It's a huge clusterfuck.

File: ntoskrnl/cache/mdlsup.c [hunk 4]
```diff
@@ -83,7 +83,7 @@ NTAPI
 CcMdlReadComplete2(IN PFILE_OBJECT FileObject,
                    IN PMDL MdlChain)
 {
-    UNIMPLEMENTED
+	UNIMPLEMENTED;
```
Comment (binarymaster) [inline ntoskrnl/cache/mdlsup.c]: There should be 4 spaces instead of Tab.

File: ntoskrnl/cache/mdlsup.c [hunk 5]
```diff
@@ -112,7 +112,7 @@ CcMdlWriteComplete2(IN PFILE_OBJECT FileObject,
                     IN PLARGE_INTEGER FileOffset,
                     IN PMDL MdlChain)
 {
-    UNIMPLEMENTED
+	UNIMPLEMENTED;
```
Comment (binarymaster) [inline ntoskrnl/cache/mdlsup.c]: Indentation again.

File: ntoskrnl/mm/section.c [hunk 6]
```diff
@@ -4814,7 +4814,10 @@ MmFlushImageSection (IN PSECTION_OBJECT_POINTERS SectionObjectPointer,
         if (SectionObjectPointer->ImageSectionObject)
         {
             DPRINT1("SectionObject has ImageSection\n");
-            return FALSE;
+			if (SectionObjectPointer->DataSectionObject) {
```
Comment (binarymaster) [inline ntoskrnl/mm/section.c]: Problem with tab's here too.

File: ntoskrnl/cache/copysup.c [hunk 7]
```diff
@@ -121,7 +121,9 @@ CcFastCopyRead(IN PFILE_OBJECT FileObject,
                OUT PVOID Buffer,
                OUT PIO_STATUS_BLOCK IoStatus)
 {
-    UNIMPLEMENTED_DBGBREAK();
+	LARGE_INTEGER Offset;
+	Offset.QuadPart = FileOffset;
+	CcCopyRead(FileObject, &Offset, Length, FALSE, Buffer, IoStatus);
```
Comment (ThFabba) [inline ntoskrnl/cache/copysup.c]: This should use TRUE for the Wait parameter. Check out the already existing version: https://github.com/reactos/reactos/blob/master/ntoskrnl/cc/copy.c#L464-L490

File: ntoskrnl/mm/section.c (hunk 6 above)
Comment (ThFabba) [reply ntoskrnl/mm/section.c]: And brace position

File: .gitignore [hunk 8]
```diff
@@ -1,2 +1,5 @@
 output-*
 modules/optional
+/.vs
+/x86build
+/CMakeSettings.json
```
Comment (ThFabba) [inline .gitignore]: This change seems unrelated. If you think it's necessary, please put it in another PR and explain it there.

Comment (woachk) [reply .gitignore]: Was added when using the Visual Studio IDE automatically...

File: ntoskrnl/cache/mdlsup.c [hunk 9]
```diff
@@ -83,7 +83,7 @@ NTAPI
 CcMdlReadComplete2(IN PFILE_OBJECT FileObject,
                    IN PMDL MdlChain)
 {
-    UNIMPLEMENTED
+	CcMdlReadComplete(FileObject, MdlChain);
```
Comment (ThFabba) [inline ntoskrnl/cache/mdlsup.c]: This is acceptable as a stub, but note that the current Cc has a correct implementation for all four of these.

Comment (woachk) [reply ntoskrnl/cache/mdlsup.c]: The current Cc doesn't seem to have correct implementations I'm afraid...

Comment (ThFabba) [reply ntoskrnl/cache/mdlsup.c]: They're much closer to correct than NewCc's.

File: ntoskrnl/cache/mdlsup.c [hunk 10]
```diff
@@ -112,7 +112,7 @@ CcMdlWriteComplete2(IN PFILE_OBJECT FileObject,
                     IN PLARGE_INTEGER FileOffset,
                     IN PMDL MdlChain)
 {
-    UNIMPLEMENTED
+	CcMdlWriteComplete2(FileObject, FileOffset, MdlChain);
```
Comment (HeisSpiter) [inline ntoskrnl/cache/mdlsup.c]: Unless I misread something, that'll just loop till end of stack?

**Comment by woachk** [comment]
Closing for some reasons...

**Comment by binarymaster** [comment]
> Closing for some reasons...

Which reasons? :thinking:
